spec.bs

```diff
@@ -950,10 +951,10 @@ To <dfn>validate fetching response</dfn> given a [=response=] |response| and |re
 </div>
 
 <div algorithm>
-To <dfn>fetch script</dfn> given a [=URL=] |url|:
+To <dfn>fetch script</dfn> given a [=URL=] |url|, and an [=origin=] |frameOrigin|:

```

Comment:
So these requests use your (and Fetch's) default of `no-cors` request mode, but explicitly pass in a `frameOrigin` here, even though in Fetch, we seem to never append an `Origin` header for no-cors requests (https://fetch.spec.whatwg.org/#append-a-request-origin-header). So I think this change does not buy us anything. Can you help me understand what the implementation does here regarding initiator origin and request mode and maybe we can work backward from there to see what we need here?

In general we should have a pretty good reason to not use CORS, and we might have one, but we _really_ need to document it in the spec.

Reply:
[Our code](https://source.chromium.org/chromium/chromium/src/+/refs/heads/main:content/browser/interest_group/auction_url_loader_factory_proxy.cc;drc=ac12f6366c92c4f41c06e2005598b1bc54b91c25;l=154) sets [**"request_initiator"**](https://crsrc.org/c/services/network/public/mojom/url_request.mojom;drc=5992439d25f71ce29efa8db1c699b99e8773d41f;l=183) to frame origin, and that's what I wanted to set here. I think I didn't find the correct fetch request field for this (thought origin was the closest).

A [fetch request's initiator](https://fetch.spec.whatwg.org/#concept-request-initiator) seems to be a different thing from the request_initiator field.

Reply:
Also, these requests [explicitly set](https://source.chromium.org/chromium/chromium/src/+/refs/heads/main:content/browser/interest_group/auction_url_loader_factory_proxy.cc;drc=ac12f6366c92c4f41c06e2005598b1bc54b91c25;l=170) `no-cors` request mode.

Reply:
> A [fetch request's initiator](https://fetch.spec.whatwg.org/#concept-request-initiator) seems to be a different thing from the request_initiator field as well.

Right. The equivalent is https://fetch.spec.whatwg.org/#concept-request-origin.

Thanks for the code links. I'm concerned about:

> For seller worklets, while the publisher page provides both the script
> and the trusted signals URLs, both requests use safe methods (GET), and
> don't set any headers, so CORS is not needed. CORB would block the

I'm reading this as "because these requests would not trigger a preflight request, CORS is not needed". I'm personally unconvinced by this. CORS explicitly has a different surface area than CORS preflight requests, so not triggering the latter is not a prerequisite for avoiding the former. I think this needs a proper security review (apologies if one was already done for this and I missed it).

Reply:
I'm not aware of such a security review.
Adding @JensenPaul